### ISSUE TITLE
Implement embedding prediction feedback loop

### ISSUE THREAD

**Issue by tzervas** [tzervas/tritter#59]
## Overview

Implement the core embedding prediction paradigm where model outputs embeddings that are fed back as inputs, rather than discrete token logits.

## Background

The Coconut/LCM-style architecture operates in continuous embedding space:
1. Input tokens → embeddings
2. Transformer processes embeddings
3. Output: next embedding (not token logits)
4. Output embedding fed back as next input

This enables latent reasoning without token overhead.

## Requirements

- [ ] Implement hidden state feedback mechanism
- [ ] Add `embedding` attention mode (bidirectional for latent processing)
- [ ] Implement curriculum: start with token prediction, gradually shift to embedding
- [ ] Add KNN/VQ rounding for converting embeddings back to tokens when needed
- [ ] Create evaluation metrics for embedding prediction quality

## Architecture Changes

```python
# Current (token prediction)
logits = model(input_ids)  # (B, L, vocab_size)
next_token = logits.argmax(-1)

# Target (embedding prediction)  
next_embedding = model(input_embeddings)  # (B, L, hidden_size)
# Feed back: input_embeddings = concat(input_embeddings, next_embedding)
```

## References

- SPEC-003-embedding-prediction.md
- ADR-001-sequence-position-vs-token-semantics.md
- Coconut paper, LCM paper

## Phase

Phase 5: Embedding Prediction (Core Innovation)
